src/meson.build

```diff
@@ -0,0 +1,218 @@
+    'xviewer-job-scheduler.h',
+    'xviewer-jobs.h',
+    'xviewer-list-store.h',
+    'xviewer-properties-dialog.h',
+    'xviewer-scroll-view.h',
+    'xviewer-sidebar.h',
+    'xviewer-statusbar.h',
+    'xviewer-thumbnail.h',
+    'xviewer-thumb-nav.h',
+    'xviewer-thumb-view.h',
+    'xviewer-transform.h',
+    'xviewer-uri-converter.h',
+    'xviewer-window-activatable.h',
+    'xviewer-window.h',
+]
+
+libxviewer_sources = [
+    'xviewer-application-activatable.c',
+    'xviewer-application.c',
+    'xviewer-clipboard-handler.c',
+    'xviewer-close-confirmation-dialog.c',
+    'xviewer-debug.c',
+    'xviewer-error-message-area.c',
+    'xviewer-file-chooser.c',
+    'xviewer-image.c',
+    'xviewer-image-jpeg.c',
+    'xviewer-image-save-info.c',
+    'xviewer-jobs.c',
+    'xviewer-job-scheduler.c',
+    'xviewer-list-store.c',
+    'xviewer-metadata-details.c',
+    'xviewer-metadata-reader.c',
+    'xviewer-metadata-reader-jpg.c',
+    'xviewer-metadata-reader-png.c',
+    'xviewer-metadata-sidebar.c',
+    'xviewer-pixbuf-util.c',
+    'xviewer-plugin-engine.c',
+    'xviewer-preferences-dialog.c',
+    'xviewer-print.c',
+    'xviewer-print-image-setup.c',
+    'xviewer-print-preview.c',
+    'xviewer-properties-dialog.c',
+    'xviewer-save-as-dialog-helper.c',
+    'xviewer-scroll-view.c',
+    'xviewer-session.c',
+    'xviewer-sidebar.c',
+    'xviewer-statusbar.c',
+    'xviewer-thumbnail.c',
+    'xviewer-thumb-nav.c',
+    'xviewer-thumb-view.c',
+    'xviewer-transform.c',
+    'xviewer-uri-converter.c',
+    'xviewer-util.c',
+    'xviewer-window-activatable.c',
+    'xviewer-window.c',
+    'zoom.c'
+]
+
+# enums as .c and .h
+xviewer_enums = gnome.mkenums(
+    'xviewer-enum-types',
+    sources: public_headers,
+    c_template: 'xviewer-enum-types.c.template',
+    h_template: 'xviewer-enum-types.h.template',
+    identifier_prefix: 'Xviewer',
+    symbol_prefix: 'xviewer'
+)
+
+# enums as .xml for gschemas
+mkenums = gnome.mkenums(
+    'org.x.viewer.enums.xml',
+    sources: ['xviewer-scroll-view.h', 'xviewer-window.h'],
+    comments: '<!-- @comment@ -->',
+    fhead: '<schemalist>',
+    vhead: '  <@type@ id=\'org.x.viewer.@EnumName@\'>',
+    vprod: '    <value nick=\'@valuenick@\' value=\'@valuenum@\'/>',
+    vtail: '  </@type@>',
+    ftail: '</schemalist>',
+    install_header: true,
+    install_dir: join_paths(datadir, 'glib-2.0', 'schemas'),
+)
+
+xviewer_marshal = gnome.genmarshal(
+    'xviewer-marshal',
+    sources: 'xviewer-marshal.list',
+    prefix: 'xviewer_marshal'
+)
+
+xviewer_resources = gnome.compile_resources(
+    'xviewer-resources',
+    'resources/xviewer.gresource.xml',
+    source_dir: 'resources',
+    c_name: '_xviewer'
+)
+
+xviewer_deps = [
+    config_h,
+    glib,
+    gtk,
+    libpeas,
+    libpeas_gtk,
+    math,
+    X11,
+    zlib,
+    cinnamon_desktop,
+    xapp
+]
+
+if enable_exif
+    xviewer_deps += exif
+    libxviewer_sources += 'xviewer-exif-util.c'
```

Comment:
```meson
if exif.found()
    libxviewer_sources += 'xviewer-exif-util.c'
    public_headers += 'xviewer-exif-util.h'
endif
# this will just be a no-op if it isn't found
xviewer_deps += exif
```

You could add all the xviewer_deps in one list rather than piecemeal.

This is all possible because of using feature options.